ejb3/src/main/java/org/jboss/as/ejb3/remote/LocalEjbReceiver.java

```diff
@@ -223,7 +225,16 @@ protected void processInvocation(final EJBClientInvocationContext invocation, fi
                     protected Object runInvocation() throws Exception {
                         setSecurityContextOnAssociation(securityContext);
                         try {
+                            Object result = view.invoke(interceptorContext);
+                            // WFLY-4331 - clone the result of an async task
+                            if(result instanceof AsyncResult) {
+                                  Object asyncValue = ((AsyncResult)result).get();
+                              return new AsyncResult(LocalEjbReceiver.clone(asyncValue.getClass(), resultCloner, asyncValue, allowPassByReference));
+                            }
                             return view.invoke(interceptorContext);
```

Comment:
Otherwise invocations which were made asynchronous by way of the EJBClient async methods will fail with CCE.